substrate/frame/multi-asset-bounties/src/lib.rs

```diff
@@ -0,0 +1,1895 @@
+		///   do their job and they willingly give up. We could slash them, but for now we allow
+		///   them to recover their deposit and exit without issue. (We may want to change this if
+		///   it is abused).
+		/// - If successful, the child-/bounty status is updated to `CuratorUnassigned`. To
+		///   reactivate the bounty, a new curator must be proposed and must accept the role.
+		///
+		/// ### Parameters
+		/// - `parent_bounty_id`: Index of parent bounty.
+		/// - `child_bounty_id`: Index of child-bounty.
+		///
+		/// ## Events
+		///
+		/// Emits [`Event::CuratorUnassigned`] if successful.
+		#[pallet::call_index(4)]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::unassign_curator())]
+		pub fn unassign_curator(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			child_bounty_id: Option<BountyIndex>,
+		) -> DispatchResult {
+			let maybe_sender = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::RejectOrigin::ensure_origin(origin).map(|_| None))?;
+
+			let (_, _, _, status, parent_curator) =
+				Self::get_bounty_details(parent_bounty_id, child_bounty_id)?;
+
+			match status {
+				BountyStatus::Funded { ref curator } => {
+					// A bounty curator has been proposed, but not accepted yet.
+					// Either `RejectOrigin`, parent bounty curator or the proposed
+					// curator can unassign the child-/bounty curator.
+					ensure!(
+						maybe_sender.map_or(true, |sender| {
+							sender == *curator ||
+								parent_curator
+									.map_or(false, |parent_curator| sender == parent_curator)
+						}),
+						BadOrigin
+					);
+				},
+				BountyStatus::Active { ref curator, .. } => {
+					let maybe_curator_deposit =
+						CuratorDeposit::<T, I>::take(parent_bounty_id, child_bounty_id);
+					// The child-/bounty is active.
+					match maybe_sender {
+						// If the `RejectOrigin` is calling this function, burn the curator deposit.
+						None => {
+							if let Some(curator_deposit) = maybe_curator_deposit {
+								T::Consideration::burn(curator_deposit, curator);
+							}
+							// Continue to change bounty status below...
+						},
+						Some(sender) if sender == *curator => {
+							if let Some(curator_deposit) = maybe_curator_deposit {
+								// This is the curator, willingly giving up their role. Free their
+								// deposit.
+								T::Consideration::drop(curator_deposit, curator)?;
+							}
+							// Continue to change bounty status below...
+						},
+						Some(sender) => {
+							if let Some(parent_curator) = parent_curator {
+								// If the parent curator is unassigning a child curator, that is not
+								// itself, burn the child curator deposit.
+								if sender == parent_curator && *curator != parent_curator {
+									if let Some(curator_deposit) = maybe_curator_deposit {
+										T::Consideration::burn(curator_deposit, curator);
+									}
+								} else {
+									return Err(BadOrigin.into());
+								}
+							}
```

Comment:
isn't this missing another ` else { return Err(BadOrigin.into()); }` ? if the sender is some, but it is not the curator but the bounty has no parent curator then it is a bad origin.

Reply:
I see for a regular bounty with no parent, the parent curator is itself. I guess it is ok.